lib/instrumentation/modules/amqplib.js

```diff
@@ -0,0 +1,54 @@
+'use strict'
+
+var shimmer = require('../shimmer')
+var { getDBDestination } = require('../context')
+
+var URL = require('url-parse')
+
+module.exports = function (amqplib, agent, { version, enabled }) {
+  agent.logger.debug('shimming amqplib.connect')
+  shimmer.wrap(amqplib, 'connect', wrapConnect)
+
+  return amqplib
+
+  function patchChannelModel (channelModel, host, port) {
+    agent.logger.debug('shimming amqplib.ChannelModel.connection.sendMessage')
+    shimmer.wrap(channelModel.connection, 'sendMessage', function (orig) {
+      return function (channel, method, fields, properties, props, content) {
+        var span = enabled && agent.startSpan(null, 'messaging', 'amqp', 'publish')
+        var id = span && span.transaction.id
+
+        agent.logger.debug('intercepted call to amqplib.Connection.sendMessage %o', { id: id, channel: channel, content: content })
+
+        if (span) {
+          span.setDestinationContext(getDBDestination(span, host, port))
+        }
+
+        var res = orig.apply(this, arguments)
+        span.end()
+        return res
+      }
+    })
+  }
+
+  function wrapConnect (original) {
+    return function wrappedConnect (url) {
+      let protocol, host, port
+      if (typeof url === 'object') {
+        protocol = (url.protocol || 'amqp') + ':'
+        host = url.hostname
+        port = url.port || ((protocol === 'amqp:') ? 5672 : 5671)
```

Comment:
nit: you can move the default port logic inside here to avoid doing two times
https://github.com/elastic/apm-agent-nodejs/blob/7d1c43b1986f112c686a4a0c0c95f5c7640a8405/lib/instrumentation/modules/amqplib.js#L24

Reply:
Fixed in https://github.com/elastic/apm-agent-nodejs/pull/1792/commits/38fdf0b00bc64ff5933f58c194a74a4b54765f88